src/test/assembly/62531-array-cmp.rs

```diff
@@ -0,0 +1,21 @@
+// Ensure the asm for array comparisons is properly optimized on x86.
+
+// min-llvm-version: 11
+// assembly-output: emit-asm
+// compile-flags: -C opt-level=2
+// only-x86_64
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: compare
+// CHECK: movb $1, %al
```

Comment:
Probably should `CHECK-NEXT` here. There are a number of plausible ways to generate code for this that would result in these two instructions somewhere, but not necessarily in _just_ these two instructions.